sdk/identity/Azure.Identity/samples/TokenCache.md

```diff
@@ -1,110 +1,43 @@
-
-> IMPORTANT! This sample assumes the location of the file it is using for storage is secure. The `Serialize` and `SerializeAsync` methods will write the unencrypted content of the `TokenCache` to the provide stream. It is the responsibility the implementer to properly protect the `TokenCache` data.
-
-The `Serialize` or `SerializeAsync` methods can be used to write out content of a `TokenCache` to any writeable stream.
-
-```C# Snippet:Identity_TokenCache_CustomPersistence_Usage_TokenCachePath
-private const string TOKEN_CACHE_PATH = "./tokencache.bin";
-```
-
-```C# Snippet:Identity_TokenCache_CustomPersistence_Write
-using (var cacheStream = new FileStream(TOKEN_CACHE_PATH, FileMode.Create, FileAccess.Write))
-{
-    await TokenCacheSerializer.SerializeAsync(tokenCache, cacheStream);
-}
-```
-
-The `Deserialize` or `DeserializeAsync` methods can be used to read the content of a `TokenCache` from any readable stream.
-
-```C# Snippet:Identity_TokenCache_CustomPersistence_Read
-TokenCache tokenCache;
-
-using (var cacheStream = new FileStream(TOKEN_CACHE_PATH, FileMode.OpenOrCreate, FileAccess.Read))
-{
-    tokenCache = await TokenCacheSerializer.DeserializeAsync(cacheStream);
-}
-```
-
-Applications can combine these methods along with the `Updated` event to automatically persist and read the token from a storage solution of their choice.
-
-```C# Snippet:Identity_TokenCache_CustomPersistence_Usage
-public static async Task<TokenCache> ReadTokenCacheAsync()
-{
-    TokenCache tokenCache;
-
-    using (var cacheStream = new FileStream(TOKEN_CACHE_PATH, FileMode.OpenOrCreate, FileAccess.Read))
-    {
-        tokenCache = await TokenCacheSerializer.DeserializeAsync(cacheStream);
-        tokenCache.Updated += WriteCacheOnUpdateAsync;
-    }
-
-    return tokenCache;
-}
-
-public static async Task WriteCacheOnUpdateAsync(TokenCacheUpdatedArgs args)
-{
-    using (var cacheStream = new FileStream(TOKEN_CACHE_PATH, FileMode.Create, FileAccess.Write))
-    {
-        await TokenCacheSerializer.SerializeAsync(args.Cache, cacheStream);
-    }
-}
-
-public static async Task Main()
-{
-    var tokenCache = await ReadTokenCacheAsync();
-
-    var credential = new InteractiveBrowserCredential(
-        new InteractiveBrowserCredentialOptions { TokenCache = tokenCache }
-    );
-}
-```
+By setting `AllowUnencryptedStorage` to `true`, the `TokenCache` will encrypt the contents of the `TokenCache` before persisting it if data protection is available on the current platform, otherwise it will write and read the `TokenCache` data to an unencrypted local file ACL'd to the current account. If `AllowUnencryptedStorage` is `false` (the default) a `CredentialUnavailableException` will be raised in the case no data protection is available.
```

Comment:
This is not fully true. We will only fallback to storing the unencrypted cache if there is no OS encryption support available. If support is available the cache will be encrypted regardless of the vaule of `UnsafeAllowUnencryptedStorage`